kombu/asynchronous/hub.py

```diff
@@ -254,6 +254,22 @@ def close(self, *args):
         for callback in self.on_close:
             callback(self)
 
+        # Complete remaining todo befor Hub close
+        # Eg: Acknowledge message
+        # To avoid infinite loop where one of the callables adds items
+        # to self._ready (via call_soon or otherwise), we take pop only
+        # N items from the ready set.
+        # N represents the current number of items on the set.
+        # That way if a todo adds another one to the ready set
+        todo = self._ready
```

Comment:
Nice thought to avoid a possible infinite loop. So here we give a last change for items to be processed?